R/as_wide_table.R

```diff
@@ -48,61 +49,139 @@ as_wide_table2 <- function(austraits){
   # Getting rid of the columns that will soon be deleted in the next austraits release and renaming the description column
   austraits$methods <- 
     austraits$methods %>% 
-    dplyr::rename(c("dataset_description" = "description"))  %>% 
+    dplyr::rename(dataset_description = "description")  %>% 
     dplyr::distinct()
   
   # collapse into one column
   austraits$locations <- 
     austraits$locations %>% 
     dplyr::filter(value!="unknown") %>% 
-    dplyr::rename(c("property" = "location_property")) %>%
+    dplyr::rename("property" = "location_property") %>%
     split(., .$dataset_id) %>%
-    purrr::map_dfr(process_table2)
+    purrr::map_dfr(process_table3)
 
-  # rename taxonomic_reference field to reflect the APC/APNI name matching process better
+  # rename taxonomic_dataset field to reflect the APC/APNI name matching process better
   austraits$taxa <- 
     austraits$taxa %>% 
-    dplyr::rename(c("taxonNameValidation" = "taxonomic_reference")) %>% 
+    # dplyr::rename("taxonNameValidation" = "taxonomic_dataset") %>% 
     dplyr::distinct()
   
   austraits_wide <- 
     austraits$traits %>% 
     dplyr::left_join(by=c("dataset_id", "location_id"), austraits$locations) %>%
-    dplyr::left_join(by=c("dataset_id", "trait_name"), austraits$methods) %>%
+    dplyr::left_join(by=c("dataset_id", "method_id", "trait_name"), austraits$methods) %>%
     dplyr::left_join(by=c("taxon_name"), austraits$taxa)
 
     # reorder the names to be more intuitive
     austraits_wide %>% dplyr::select(
       
     # The most useful (if you are filtering for just one taxon_name)
-      dataset_id, observation_id, trait_name, taxon_name, value, unit, 
-      entity_type, population_id, individual_id,
-      value_type, basis_of_value, 
-      replicates, 
+      "dataset_id", "observation_id", "trait_name", "taxon_name", "value", "unit", 
+      "entity_type", "population_id", "individual_id",
+      "value_type", "basis_of_value", 
+      "replicates", 
```

Comment:
Does this not require an `all_of(c(....))`?

Reply:
From my understanding

`all_of` is only needed when the vector of character strings are outside of the select call

e.g. 

`
vars_keep <- c("trait_name", "taxon_name")
data |> select(all_of(vars_keep)) 

Here we are directly typing out what we need inside `select`. All the tidyselect warnings should have been dealt with in the previous reviews, all tests are passing so I think we are ok here :D